target/linux/ramips/image/mt7620.mk

```diff
@@ -468,6 +468,16 @@ define Device/hnet_c108
 endef
 TARGET_DEVICES += hnet_c108
 
+define Device/hootoo_tm05
+  MTK_SOC := mt7620n
+  IMAGE_SIZE := 7872k
+  DEVICE_VENDOR := HooToo
+  DEVICE_MODEL := TM05
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-i2c-core kmod-i2c-ralink
```

Comment:
i2c-core will be selected by i2c-ralink. Remove it here.

Reply:
OK, will do. I had some issues with it not being selected, but perhaps something else was wrong at the time. Will remove, let you know.

Reply:
Hmmm ... made the change, seem to have lost i2c now. But wondering if I need to reset / clean .config or something like that, to get it back.

Thoughts? Thanks!

Reply:
This is the only one I'm really struggling with (again ... LOL). I say again because it didn't seem to work a week or two back, then "magically" propagated from the image makefile to top level config. But not happening again now - any thoughts at all? Thanks!

Reply:
OK, figured out the "mystery" ... :-). If I execute the commands below (clean .config, let it "auto-build" + select only this device),
```
cat /dev/null > .config
make menuconfig

```
2 cases, without output as below (grep kmod-i2c .config),
1) Only include kmod-i2c-ralink
```
CONFIG_DEFAULT_kmod-i2c-ralink=y
# CONFIG_PACKAGE_kmod-i2c-core is not set
# CONFIG_PACKAGE_kmod-i2c-gpio-custom is not set
```
2) Include both kmod-i2c-core kmod-i2c-ralink
```
CONFIG_DEFAULT_kmod-i2c-core=y
CONFIG_DEFAULT_kmod-i2c-ralink=y
CONFIG_PACKAGE_kmod-i2c-core=y
# CONFIG_PACKAGE_kmod-i2c-algo-bit is not set
# CONFIG_PACKAGE_kmod-i2c-algo-pca is not set
# CONFIG_PACKAGE_kmod-i2c-algo-pcf is not set
# CONFIG_PACKAGE_kmod-i2c-gpio is not set
# CONFIG_PACKAGE_kmod-i2c-gpio-custom is not set
# CONFIG_PACKAGE_kmod-i2c-mux is not set
# CONFIG_PACKAGE_kmod-i2c-pxa is not set
CONFIG_PACKAGE_kmod-i2c-ralink=y
# CONFIG_PACKAGE_kmod-i2c-smbus is not set
# CONFIG_PACKAGE_kmod-i2c-tiny-usb is not set
```

So it seems I do need both. Not saying it's "right", just that it seems to be the result. Thanks!

Reply:
I have to look into it, maybe the package dependencies are not set correctly. Keep it included for now.

Reply:
Sure, will do - NP. I do remember having to mess with this to get it working. As you say, likely just a minor dependency item somewhere. Will leave it as it is, can change it later if needed. Thanks!

Reply:
Looks like i2c modules do not automatically select their dependencies. I remember a similar case a long time ago. So, just keep i2c-core included.

Reply:
NP, will do. Thanks!

Reply:
FYI, just built this way, sysupgrade => i2c working again ... :-).